cmd/common/utils/http.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+func DownloadFile(rawURL string) (io.ReadCloser, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, fmt.Errorf("parsing URL: %w", err)
+	}
+
+	client := http.Client{
+		Timeout: 2 * time.Second,
+	}
+
+	switch u.Scheme {
+	case "http", "https":
+		resp, err := client.Get(rawURL)
+		if err != nil {
+			return nil, fmt.Errorf("downloading file: %w", err)
```

Comment:
I think returning `err` is enough, since the message is already good enough on the consumer and this func is always about downloading.